resources/datadog-integrations/project.rb.erb

```diff
@@ -43,8 +43,11 @@ end
 dependency 'preparation'  # creates required build directories
 dependency 'dd-check-<%= name %>-software'
 
+# Checks should require the datadog agent
+runtime_dependency 'datadog-agent'
+
 extends_packages('datadog-agent', 'datadog')
 
-extra_package_file '/etc/dd-agent/conf.d/<%= name %>.yaml.example'
+extra_package_file '/etc/dd-agent/conf.d/examples/<%= name %>.yaml.example'
```

Comment:
I dont think we can assume users who upgrade individual checks will only be expects who understand all the config options.  I'd push for all the examples to be in a single slot to avoid confusion and support pains. If that is conf.d/examples/ thats great if its all in conf.d thats also fine. 

Many of our challenges here are coming from choosing to ship all the integrations both in the agent itself, as well as in packages. As this results in both packages wreslting over ownership of a particular file.  If the agent package only installed the agent, but depending on the individual check packages we'd simplify this greatly.